<!DOCTYPE html>
<html class="h-full" lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title >M1CR0B1AL1Z3R</title>
  <link rel="icon" href="data:;base64,iVBORw0KGgo=">
  <link rel="icon" type="image/x-icon" href="{{url_for('static',filename='images/dna.svg')}}">
  <link rel="stylesheet" href="{{url_for('static',filename='css/output.css')}}">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
  <!-- <script src="https://cdn.tailwindcss.com"></script> -->
</head>

<body class="font-sans h-full">
  <div class='w-full bg-no-repeat bg-cover bg-attachment' 
       style="background-image: url({{file_name}});">

	<div class="flex flex-col mx-auto  h-full justify-start items-center">
    <div class="flex flex-col my-4 justify-center items-center">
        <h2 class="text-9xl text-center  text-white select-none my-3 text-outline">About</h2>
		<p class="w-1/2 text-center text-1 p-5 m-8 rounded-md  text-black bg-white bg-opacity-75  select-none text-justify">
			 M1CR0B1AL1Z3R allows generating both quantitative and qualitative measurements for ad-hoc downstream analyses given a microbial genomic sequences dataset. Visual and textual results, ready for publication or further analysis, are given as an output. For example, open reading frames of each genome, orthologous groups (both unaligned and aligned), core proteome, phylogenetic tree, GC content and more. See running example <a href="https://microbializer.tau.ac.il/example_data.zip" target="_blank">input</a> and <a href="https://microbializer.tau.ac.il/example_data/result.html" target="_blank">output</a>.
		</p>
		<h2 class="text-7xl text-center  text-white select-none my-3 text-outline">Terms of Use</h2>
		<p class="w-1/2 text-center text-1 p-5 m-8 rounded-md  text-black bg-white bg-opacity-75  select-none text-justify">
			Copyright (C) 2018 <a href="https://www.tau.ac.il/~talp/" target="_blank">Tal Pupko</a>. This program is a
			free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as
			published by the Free Software Foundation, either version 2 of the License, or (at your option) any later
			version. This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
			even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
			License for more details. You should have received a copy of the GNU General Public License along with this
			program. If not, see <a href="https://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html" target="_blank">
			https://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html</a>. The beta version is available
			<a href="https://github.com/orenavram/Microbializer" target="_blank">here</a>.
		</p>
		<h2 class="text-7xl text-center  text-white select-none my-3 text-outline">Credits</h2>
		<p class="w-1/2 text-center text-1 p-5 m-8 rounded-md  text-black bg-white bg-opacity-75  select-none text-justify">
			When citing M1CR0B1AL1Z3R please refer to:<br>
			<b>M1CR0B1AL1Z3R - a user-friendly web server for the analysis of large-scale microbial genomics data</b><br>
			Avram O., Rapoport D., Portugez S., & Pupko T. (2019)<br>
			<i>Nucleic Acids Research</i> | <a href="https://doi.org/10.1093/nar/gkz423" target="_blank">https://doi.org/10.1093/nar/gkz423</a>.
			<br>
			The source code is available at:<br> <a href="https://github.com/orenavram/microbializer">https://github.com/orenavram/microbializer</a>
		</p>
		
        <h2 class="text-7xl text-center  text-white select-none my-3 text-outline">Acknowledgements</h2>
		<p class="w-1/2 text-center text-1 p-5 m-8 rounded-md  text-black bg-white bg-opacity-75  select-none text-justify">
			This study was supported in part by a fellowship from the <a href="http://safrabio.cs.tau.ac.il/">Edmond J. Safra Center for Bioinformatics</a> at Tel Aviv University and Israel Science Foundation Grant No. 1282/17.
			<a href="https://www.tau.ac.il/~orenavr2/">Oren Avram</a>, Dana Rapoport, Shir Portugez, and <a href="https://www.tau.ac.il/~talp/">Tal Pupko</a> developed the algorithmic pipeline.
			<a href="https://www.tau.ac.il/~orenavr2/">Oren Avram</a> developed the web server.
        </p>
        <p class="w-1/2 text-center text-l p-5 m-8 rounded-md  text-black bg-white bg-opacity-75 ">
            {{contact_info}}
			<br>
			Follow us on <a href="https://twitter.com/@orenavram" target="_blank"><i style="font-size:24px" class="fa fa-twitter"></i> Twitter</a> for updates!
        </p>
    </div>
    <a  href="/" class="w-64  flex flex-col  items-center  px-4  py-6  bg-white  rounded-md  shadow-lg  tracking-wide  uppercase  border border-blue-700  bg-white-600 text-black hover:bg-purple-600 hover:text-white ease-linear transition-all  duration-150  mx-8  my-4 cursor-pointer">
        back to home page
    </a>
</div>
 </div>
</body>

