{% extends 'base.html' %}

{% block body %}

<div class="flex flex-col mx-auto  h-full justify-start items-center">
    <div class="flex flex-col my-4 justify-center items-center">
        <h1 class="text-9xl text-center  text-white select-none my-5 text-outline">FAQ</h1>
		
		<h2 class="text-4xl text-center  text-white select-none my-5 text-outline">What are the tunable settings and how can they influence my results?</h2>
        <p class="w-3/4 text-center text-l p-3 m-4 rounded-md  text-black bg-white bg-opacity-75 text-justify">
            &#x25cf; Input fasta files type: The input files can be either GENOMES (fully assembled or contigs) or ORFs 
			(open reading frames). In the case of GENOMES, the first step in the pipeline is ORFs prediction using 
			<a class="text-blue-600 underline decoration-solid" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-119" target="_blank">Prodigal</a>. 
			If the user already has ORFs and would like to use them as the starting point, they should choose ORFs in this 
			parameter, and in that case ORFs prediction is skipped.
			<br><br>
			&#x25cf; Filter out contigs / orfs of plasmids: The pipeline can as a first step filter out contigs or ORFs that 
			belong to plasmids. This is done by a simple search for the word "plasmid" in the contig or ORF record name.
			<br><br>
			&#x25cf; Minimum sequence identity & sequence coverage (protein-level) for homologs detection: 
			These parameters are used in the homology search step, which is the first step in the orthogroup inference step. 
			The homology search is performed using the MMSEQS2 program, which is a fast and sensitive homology search tool. 
			The default values for these parameters are 40% sequence identity and 70% sequence coverage. 
			These values can be adjusted to increase or decrease the stringency of the homology search. 
			For example, if the user is interested in detecting more remote homologs, the sequence identity and 
			coverage thresholds should be decreased. In that case, the inferred orthogroups will be larger and contain more 
			genes in each orthogroup. In contrast, if the user is interested in detecting only very close homologs, the 
			sequence identity and coverage thresholds should be increased. In that case, the inferred orthgroups will be 
			smaller as they will contain only very similar genes.
			<br><br>
			&#x25cf; Minimum percent of strains required to consider an orthogroup as part of the core genome:
			The parameter dictates the inclusion or exclusion of orthogroups in the core proteome (and core genome). 
			By default, this value is set to 100% and thus, only orthogroups that contain members of all analyzed genomes 
			are included in the core proteome. However, when bacteria from different orders are analyzed, this strict 
			definition can lead to a very small core proteome.<br>
			In that case, the core threshold can be lowered. For example, when a 70% threshold is used, orthogroups shared 
			by at least 70% of the analyzed genomes are included in the "core" proteome. In this case, the tree will be 
			inferred using a larger dataset, albeit, with missing values. As always, the best way to study the impact 
			of tunable parameters on a specific dataset is to perform trial runs using different thresholds.
			<br><br>
			&#x25cf; Root the species tree according to this outgroup: The outgroup genome is used to root the species tree. 
			By default, no outgroup is used and the produced species tree is unrooted. Alternatively, the user can indicate one 
			of the file names (without the file extension) in the uploaded dataset as an outgroup. In that case, The outgroup genome 
			should be a genome that is phylogenetically distant from the ingroup genomes.<br>
			Note that if the outgroup name is not found in the dataset, it will be ignored and the produced tree will be unrooted.
			<br><br>
			&#x25cf; Apply bootstrap over the species tree: The bootstrap values quantify the reliability of all non-trivial 
			splits (branches) in the inferred tree. By default, they are not computed. The user can choose to compute them 
			and present them on the species tree.
			<br><br>
			&#x25cf; Add orphan genes to orthogroups table: By default, orphan genes (genes that do not belong to any orthogroup)
			are not included in the orthogroups table (in the result folder 05a_orthogroups). The user can choose to include them
			as orthogroups with single genes. Of note, orthogroups that contain multiple genes of a single genome - are
			always included in the orthogroups table, regardless of this parameter.
			<br><br>
        </p>
        
		<a name="tunable_parameters"></a>
        <h2 class="text-4xl text-center  text-white select-none my-5 text-outline">Should I include an outgroup genome in my analysis?</h2>
        <p class="w-3/4 text-center text-l p-3 m-4 rounded-md  text-black bg-white bg-opacity-75 text-justify">
            An outgroup genome allows reconstructing rooted trees. However, including an outgroup genome may strongly affect the 
			results. For example, core genes are genes shared by all analyzed genomes. A phylogenetically distant outgroup may 
			share only some of the genes shared by all ingroup genomes, thus resulting in sparse data, i.e., a core proteome 
			composed of fewer genes. This, in turn, may produce a less accurate phylogenetic tree. Note that it can also lead 
			to no tree at all, if including the outgroup results in no genes being shared by all genomes
			(see <a class="text-blue-600 underline decoration-solid" href="#tree_inference">Why a species tree was not included in my analysis results?</a>).
			Including a remote outgroup may also introduce a long-branch attraction artifact, as well as biases due to different 
			nucleotide composition in the ingroup vs. outgroup sequences
			(see <a class="text-blue-600 underline decoration-solid" href="#msa_reliability">Can I trust the obtained core gene phylogenetic tree?</a>).
			Thus, in cases where an outgroup is available, we recommend running the analysis twice - with and without the outgroup. 
			The user is advised to compare these two runs and study the impact of including the outgroup on the specific data 
			being analyzed.
			<br>
        </p>
        
        <!-- <h2 class="text-4xl text-center  text-white select-none my-5 text-outline">What is the expected running time for my data?</h2>
        <p class="w-3/4 text-center text-l p-3 m-4 rounded-md  text-black bg-white bg-opacity-75 text-justify">
            The running time depends on the number of genomes, the distance between pairs of genomes, the number of ORFs, and the 
			number of core genes. It also depends on the number of jobs currently running on our computer cluster. Thus, 
			we cannot a-priori accurately estimate running times. However, a rough estimate can be obtained with the 
			following table, which details the running times of various datasets.
			<br><br>
			<img src="{{url_for('static',filename='images/running_times.png')}}" style="height: 100%; width: 100%;" id="flow">
			*Minimal percentage for core: 50% (only three genes that are shared by 100% were detected)
			<br>
        </p> -->

        <a name="homolog_inference"></a>
        <h2 class="text-4xl text-center  text-white select-none my-5 text-outline">How reliable is the orthogroup inference step and is there a way to improve it?</h2>
        <p class="w-3/4 text-center text-l p-3 m-4 rounded-md  text-black bg-white bg-opacity-75 text-justify">
			The detection of orthologous genes relies on the correct identification of ORFs. Thus, errors in detecting ORFs 
			can propagate to erroneous detection of orthologs. In addition, the ortholog-identification step relies on homology 
			search algorithms such as BLAST (or in our case, MMSEQS2). These algorithms are approximate and may lead to two 
			types of error: the identification of an erroneous orthologs and the misidentification of a correct ortholog. 
			An ortholog can also be missed because some of the analyzed genomes may not be fully assembled. Finally, 
			horizontal gene transfer (HGT) is a major evolutionary force shaping bacterial evolution and thus, seemingly 
			orthologous sequences can, in fact, represent xenology rather than orthology (xenology is when sequence similarity 
			stems from HGT events rather than from vertical divergence following speciation events). It is critically important 
			to be aware of these potential biases when interpreting the results. An excellent reference that discusses these 
			problems and suggests algorithms to test (and sometimes correct) for such biases is Philippe H., et al. 
			(<a class="text-blue-600 underline decoration-solid" href="https://doi.org/10.1371/journal.pbio.1000602" target="_blank"><i>PLoS Biology</i>, 2011</a>).
			<br>
        </p>
        
		<a name="msa_reliability"></a>
        <h2 class="text-4xl text-center  text-white select-none my-5 text-outline">Can I trust the multiple sequence alignments of the orthogroups?</h2>
        <p class="w-3/4 text-center text-ssm p-3 rounded-md  text-black bg-white bg-opacity-75 text-justify">
			Despite substantial advances in multiple alignment theory and the development of even faster and more accurate 
			alignment programs, generated alignments are still fraught with errors
			(<a class="text-blue-600 underline decoration-solid" href="https://doi.org/10.1371/journal.pone.0018093" target="_blank">Thompson JD., et al., <i>PLoS One</i>, 2011</a>; 
			<a class="text-blue-600 underline decoration-solid" href="https://doi.org/10.1093/nar/gkv318" target="_blank">Sela I., et al.,<i>Nucleic Acids Res</i>, 2015</a>). 
			In this web server, we use the MAFFT program
			(<a class="text-blue-600 underline decoration-solid" href="https://doi.org/10.1093/molbev/mst010" target="_blank">Katoh K. and Standley DM., <i>Mol Biol Evol</i>, 2013</a>), 
			which combines accuracy and computational speed, and is one of the most widely used alignment methods. 
			However, other excellent alignment programs exist. For example, the PRANK program
			(<a class="text-blue-600 underline decoration-solid" href="https://doi.org/10.1126/science.1158395" target="_blank">Loytynoja A. and Goldman N., <i>Science</i>, 2008</a>) 
			is considerably slower than MAFFT, but has been shown to yield more accurate alignments compared to MAFFT in simulation 
			studies (e.g., <a class="text-blue-600 underline decoration-solid" href="https://doi.org/10.1093/nar/gkv318" target="_blank">Sela I., et al.,<i>Nucleic Acids Res</i>, 2015</a>). 
			In our web server, we provide the inferred multiple sequence alignments for all the orthogroups, and the user can 
			download these files and realign them using any other alignment method. In addition, we suggest testing the 
			reliability of any specific alignment using existing tools, such as the 
			<a class="text-blue-600 underline decoration-solid" href="https://guidance.tau.ac.il" target="_blank">GUIDANCE2 web server</a> 
			(<a class="text-blue-600 underline decoration-solid" href="https://doi.org/10.1093/nar/gkv318" target="_blank">Sela I., et al.,<i>Nucleic Acids Res</i>, 2015</a>).
			<br>
        </p>

        <h2 class="text-4xl text-center  text-white select-none my-5 text-outline">Can I trust the produced species phylogenetic tree?</h2>
        <p class="w-3/4 text-center text-l p-3 m-4 rounded-md  text-black bg-white bg-opacity-75 text-justify">
            Reconstructing accurate phylogenetic trees is one of the "holy grails" of molecular evolution research. 
			It is a notoriously difficult task, known to be affected by many factors, including
			(i) the quality of the input sequences; (ii) the sequence sampling; (iii) the quality of the input alignment;
			(iv) the identification of orthologous sequences; (v) missing data; (vi) the assumed evolutionary model;
			(vii) the level of sequence divergence and saturation; (viii) random stochastic factors.
			Often, reconstructed trees are highly supported, yet they reflect non-phylogenetic signal rather than genuine 
			phylogenetic signal
			(<a class="text-blue-600 underline decoration-solid" href="https://doi.org/10.1371/journal.pbio.1000602" target="_blank">Philippe H, et al., <i>PLoS Biology</i>, 2011</a>). 
			For example, long tree branches tend to cluster together regardless of their true evolutionary relationships, a phenomenon 
			called "long-branch attraction"
			(<a class="text-blue-600 underline decoration-solid" href="https://doi.org/10.2307/2412923" target="_blank">Felsenstein J, <i>Syst Zool</i>, 1978</a>). 
			Similarly, and especially when bacterial sequences are analyzed, genomes with similar GC content can cluster together, 
			generating trees that reflect similarity in nucleotide composition rather than true vertical inheritance
			(<a class="text-blue-600 underline decoration-solid" href="https://doi.org/10.1073/pnas.92.24.11317" target="_blank">Galtier N and Gouy M, <i>PNAS</i>, 1995</a>). 
			Furthermore, horizontal gene transfer events can bias the inference of orthologous sequences
			(see <a class="text-blue-600 underline decoration-solid" href="#homolog_inference">How reliable is the orthogroup inference step and is there a way to improve it?</a>),
			and thus trees inferred from a concatenation of many multiple sequence alignments may reflect "an average" of 
			conflicting gene trees. This average may or may not reflect the desired "vertical" (or species) tree. 
			Eliminating these biases is an active research area. We highly recommend manually inspecting every step of the 
			M1CR0B1AL1Z3R pipeline, including the phylogenetic tree reconstruction. Hence, a user can download the core proteome 
			and try to run it with or without a specific subset of alignments. A user can test each gene for congruency with 
			the obtained species tree. Genes that are significantly incongruent with the species tree are suspected of reflecting 
			cases of horizontal transfer. One can then reconstruct the tree, without genes suspected of horizontal transfer, 
			and compare it to the tree obtained using all of the data. Note that removing genes reduces the noise, but also the 
			phylogenetic signal, and there is no consensus regarding the optimal strategy.
			We recommend Anisimova, et al. (<a class="text-blue-600 underline decoration-solid" href="https://doi.org/10.1186/1471-2148-13-161" target="_blank"><i>BMC Evolutionary Biology</i>, 2013</a>) 
			and Philippe et al. (<a class="text-blue-600 underline decoration-solid" href="https://doi.org/10.1371/journal.pbio.1000602" target="_blank"><i>PLoS Biology</i>, 2011</a>) 
			for further reading about biases and potential solutions when reconstructing phylogenetic trees.
			<br>
        </p>
		
        <h2 class="text-4xl text-center  text-white select-none my-5 text-outline">Why does the species tree reconstruction step take so long?</h2>
        <p class="w-3/4 text-center text-l p-3 m-4 rounded-md  text-black bg-white bg-opacity-75 text-justify">
            We aim to reconstruct the phylogenetic tree based on state-of-the-art methodologies. To this end, we apply the 
			maximum-likelihood paradigm, which relies on an explicit evolutionary model of sequence evolution. Specifically, 
			we assume the GTR+I+gamma model, with among-site rate variation modeled using the discrete gamma distribution. 
			The tree is reconstructed using one of the fastest programs, IQ-TREE
			(<a class="text-blue-600 underline decoration-solid" href="https://academic.oup.com/mbe/article/37/5/1530/5721363" target="_blank">Minh, Bui Quang, et al, Molecular biology and evolution, 2020</a>). 
			Despite the speed of IQ-TREE, the number of sequences and the number of sites in the alignment can significantly affect the running time.
			<br>
        </p>
		
		<a name="tree_inference"></a>
		<h2 class="text-4xl text-center  text-white select-none my-5 text-outline">Why was a species tree not included in my analysis results?</h2>
        <p class="w-3/4 text-center text-l p-3 m-4 rounded-md  text-black bg-white bg-opacity-75 text-justify">
            There are several scenarios that can lead to results lacking a phylogenetic tree. First, when a dataset consists 
			of less than four different genomic sequences, there is only one unrooted tree. Hence, there is no point in 
			running the tree search algorithm. Second, there may be no core genes. The tree reconstruction is based on the 
			core proteome, and if no core genes exist (genes shared by all genomes), there are no data to reconstruct the tree. 
			An empty core proteome could result from extensive missing data, so that each gene is missing in at least one genome. 
			A way around this is to change the definition of core genes in the Advanced Setting "Minimum percent of strains 
			required to consider an orthogroup as part of the core genome" 
			(see <a class="text-blue-600 underline decoration-solid" href="#">What are the tunable settings and how can they influence my results?</a>).
			Third, even when more than three sequences are available and core genes do exist, the tree can still be missing. 
			This can reflect genomic sequences that are identical. IQ-TREE reduces the input alignment to unique sequences, i.e., 
			it removes duplicates. If removing duplicates leaves less than four sequences, a tree will not be generated.
			<br>
        </p>	
    </div>
	<div>
		<a  href="/" class="w-64  flex flex-col  items-center  px-4  py-6  bg-white  rounded-md  shadow-lg  tracking-wide  uppercase  border border-blue-700  bg-white-600 text-black hover:bg-lime-600 hover:text-white ease-linear transition-all  duration-150  mx-8  my-4 cursor-pointer">
			back to home page
		</a>
	</div>
</div>

{% endblock %}